### PR TITLE
refactor: update expect() parse function

### PR DESCRIPTION
**Pull request by IlyaFaer**
There is a couple of issues related to the parsing functions. Trying to reduce their complexity first before starting issues.

### CODE REVIEW

File: google/cloud/spanner_dbapi/parser.py [hunk 1]
```diff
@@ -150,38 +149,29 @@ def parse_values(stmt):
 
 
 def expect(word, token):
-    word = word.strip(" ")
+    word = word.strip()
```
Comment (IlyaFaer) [inline google/cloud/spanner_dbapi/parser.py:152]: `strip()` erases whitespaces by default

File: google/cloud/spanner_dbapi/parser.py [hunk 2]
```diff
@@ -150,38 +149,29 @@ def parse_values(stmt):
...
 
             word, arg = expect(word, ARGS)
             all_args.append(arg)
+            word = word.strip()
 
-            word = word.strip(" ")
-            if word == "":
-                break
-            elif word[0] != ",":
+            if word and not word.startswith(","):
```
Comment (IlyaFaer) [inline google/cloud/spanner_dbapi/parser.py:168]: As `while` circle already checks that `word` is not empty, we can reduce number of conditions here

File: google/cloud/spanner_dbapi/parser.py [hunk 3]
```diff
@@ -150,38 +149,29 @@ def parse_values(stmt):
...
-                word = word[1:]
+            word = word[1:]
         return "", values(all_args)
 
-    elif token == TERMINAL:
-        word = word.strip(" ")
-        if word != "%s":
-            raise ProgrammingError("TERMINAL: `%s` is not %%s" % word)
-        return "", pyfmt_str
-
```
Comment (IlyaFaer) [inline google/cloud/spanner_dbapi/parser.py:184]: I didn't find any call to `expect` with `TERMINAL` token, so it looks unused

File: tests/spanner_dbapi/test_parser.py [hunk 4]
```diff
@@ -21,35 +20,6 @@
...
-            ("%sa", "TERMINAL: `%sa` is not %s"),
-        ]
-
-        for text, wantException in cases:
-            with self.subTest(text=text):
-                self.assertRaisesRegex(
-                    ProgrammingError,
-                    wantException,
-                    lambda: expect(text, TERMINAL),
-                )
```
Comment (IlyaFaer) [inline tests/spanner_dbapi/test_parser.py:51]: As `TERMINAL` token doesn't seem to be used anywhere, we don't need tests for it

**Review by mf2199 — APPROVED** [review]

**Pull request merged**